CheckDrive.Mobile/CheckDrive.Mobile/CheckDrive.Mobile/Services/ApiClient.cs

```diff
@@ -79,5 +79,41 @@ public async Task<HttpResponseMessage> PostAsync(string resource, string body)
                 throw;
             }
         }
+        public async Task<HttpResponseMessage> PutAsync(string resource, string body)
+        {
+            try
+            {
+                string token = await LocalStorage.GetAsync<string>(LocalStorageKey.Token);
+
+                if (string.IsNullOrEmpty(token))
+                {
+                    throw new InvalidTokenException("Token is empty.");
+                }
```

Comment:
It looks like these lines and line **94** are good candidate for a separate method.